src/Command/HttpSmokeTestCrawlerCommand.php

```diff
@@ -0,0 +1,167 @@
+            $this->sendEmailReport($baseUrl, $emailList, $filters, $crawlConfiguration);
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * @param string $baseUrl
+     * @param array $emails
+     * @param string $filters
+     * @param CrawlerConfiguration $crawlerConfiguration
+     */
+    private function sendEmailReport(string $baseUrl, array $emails, string $filters, CrawlerConfiguration $crawlerConfiguration)
+    {
+        $emailBody = $this->templateRenderer->make('Email/CrawlingReport', [
+            'data' => [
+                'baseUrl' => $baseUrl,
+                'userAgent' => $crawlerConfiguration->getUserAgent() ?? 'SmokeTestCrawler/1.0',
+                'filters' => $filters ?? 'no filters used',
+                'maxCrawlCount' => $crawlerConfiguration->getMaximumCrawlCount() ?? 'no limits',
+                'maxCrawlDepth' => $crawlerConfiguration->getMaximumCrawlDepth() ?? 'no limits',
+                'maxResponseSize' => $crawlerConfiguration->getMaximumResponseSize() ?? 'no limits'
+            ]
+        ])->render();
+
+        $this->crawlHandler->sendEmailReport(
+            'noreply@webcrawler.eset.com',
+            'Http Smoke Test Report (' . $baseUrl . ')',
+            $emails,
+            $emailBody
+        );
+    }
+
+    /**
+     * @param int|null $delayBetweenRequests
+     * @param bool $respectRobots
+     * @param bool $rejectNoFollowLinks
+     * @param string|null $userAgent
+     * @param int|null $maxCrawlCount
+     * @param int|null $maxCrawlDepth
+     * @param int|null $maxResponseSize
+     * @return CrawlerConfiguration
+     */
+    private function createCrawlConfigurationFromOptions(
+        ?int $delayBetweenRequests,
+        bool $respectRobots,
+        bool $rejectNoFollowLinks,
+        ?string $userAgent,
+        ?int $maxCrawlCount,
+        ?int $maxCrawlDepth,
+        ?int $maxResponseSize
+    ): CrawlerConfiguration
+    {
+        $crawlConfiguration = new CrawlerConfiguration();
+
+        if ($delayBetweenRequests !== null) {
+            $crawlConfiguration->setDelayBetweenRequests($delayBetweenRequests);
+        }
+
+        $crawlConfiguration->setRespectRobots($respectRobots);
+        $crawlConfiguration->setRejectNoFollowLinks($rejectNoFollowLinks);
+
+        if (!empty($userAgent)) {
+            $crawlConfiguration->setUserAgent($userAgent);
+        }
+
+        if (!empty($maxCrawlCount)) {
+            $crawlConfiguration->setMaximumCrawlCount($maxCrawlCount);
+        }
+
+        if (!empty($maxCrawlDepth)) {
+            $crawlConfiguration->setMaximumCrawlDepth($maxCrawlDepth);
+        }
+
+        if (!empty($maxResponseSize)) {
+            $crawlConfiguration->setMaximumResponseSize($maxResponseSize);
+        }
+
+        return $crawlConfiguration;
+    }
+
+    /**
+     * @param string|null $inputData
+     * @return array
+     */
+    private function createTrimmedArray(?string $inputData): array
```

Comment:
methods like this should be encapsulated into collection, it is this their responsibility